engine/src/main/java/org/hibernate/validator/engine/HibernateConstraintViolation.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.engine;
+
+import javax.validation.ConstraintViolation;
+
+/**
+ * A custom {@link ConstraintViolation} which allows to get additional information for a constraint violation.
+ *
+ * @since 5.3
+ */
+public interface HibernateConstraintViolation<T> extends ConstraintViolation<T> {
+	/**
+	 * @return an object that further describes the violation or {@code null}.
+	 */
+	Object getViolationContext();
```

Comment:
i also think that the term payload would be quiet confusing because of the use in the annotation. but it is true that the word context might be not that accurate as well.

how about ```getViolationDetails()``` or ```getViolationReference()``` ?